tutorials/cp-cf-understand-application-lifecycle/cp-cf-understand-application-lifecycle.md

```diff
@@ -0,0 +1,97 @@
+---
+title: Understand the Cloud Foundry Application Lifecycle for Buildpack Applications
+description: Understand the application lifecycle in Cloud Foundry and what happens when you cf push.
+auto_validation: true
+time: 15
+tags: [ tutorial>beginner, software-product>sap-btp--cloud-foundry-environment]
+primary_tag: software-product>sap-btp--cloud-foundry-environment
+author_name: Beyhan Veli
+author_profile: https://github.com/beyhan
+---
+
+## Prerequisites
+ - **Tutorials** [Fundamentals of the SAP BTP, Cloud Foundry environment](cp-cf-fundamentals)
+ - **Tutorials** [Install the Cloud Foundry Command Line Interface (CLI)](cp-cf-download-cli)
+
+
+## Details
+### You will learn
+  - What happens when you CF Push
+  - The lifecycle of buildpack applications
+  - Differences between restarting, re-staging, and re-deploying
+  - How Cloud Foundry handles crashes and shutdowns, and the process of evacuation
+
+---
+[ACCORDION-BEGIN [Step 1: ](CF Push)]
+
+When you push an application to Cloud Foundry (`cf push`), by default it uses a [buildpack](https://docs.cloudfoundry.org/buildpacks/) (either specified by you or auto-detected by Cloud Foundry) to download required dependencies, compile everything needed for your application, and produce a [droplet](https://docs.cloudfoundry.org/concepts/glossary.html) - an artifact - that can be used to run your application in a container. A `cf push` goes through three phases:
+
+  1. _Upload_ - A record is created in Cloud Foundry for the application, source files are uploaded, and an app package is created
+
+  2. _Staging_ - The required [buildpack](https://docs.cloudfoundry.org/buildpacks/) is downloaded and a droplet is created (using the generated buildpack and app package)
+
+  3. _Startup_ - The app is scheduled to run and is started up (using the droplet) within a container
+
+You can see the full application lifecycle documented at [How Apps are Staged](https://docs.cloudfoundry.org/concepts/how-applications-are-staged.html). Using the `manifest.yml` file, you can specify what actually happens during each phase, such as what buildpacks to use, the command to be run on startup, and more. A list of all the possible manifest settings can be found in the [official documentation](https://docs.cloudfoundry.org/devguide/deploy-apps/manifest-attributes.html).
+
+To see each phase in action, you can push the [cf-nodejs](https://github.com/SAP-samples/cf-sample-app-nodejs) app you worked with before. Make sure to change [random-route](https://github.com/SAP-samples/cf-sample-app-nodejs/blob/8c5da6ab1b56323a9febbc67216a12c58f1c784f/manifest.yml#L6) to `true` in the `manifest.yml` or have a unique name for the app by changing the [app name](https://github.com/SAP-samples/cf-sample-app-nodejs/blob/8c5da6ab1b56323a9febbc67216a12c58f1c784f/manifest.yml#L3) in the `manifest.yml`, and then run:
```

Comment:
Do you need to tell the user where to run the commands, or do we assume they know that?

Reply:
This feedback has been applied.